lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -179,13 +180,45 @@ export const migrateGranuleAndFilesViaTransaction = async (
 
 export const migrateGranulesAndFiles = async (
   env: NodeJS.ProcessEnv,
-  knex: Knex
+  knex: Knex,
+  granuleSearchParams: GranuleDynamoSearchParams = {}
 ): Promise<GranulesAndFilesMigrationSummary> => {
   const granulesTable = envUtils.getRequiredEnvVar('GranulesTable', env);
 
-  const searchQueue = new DynamoDbSearchQueue({
+  const defaultSearchParams = {
     TableName: granulesTable,
-  });
+  };
+  let extraSearchParams = {};
+
+  type searchType = 'scan' | 'query';
+  let dynamoSearchType: searchType = 'scan';
+
+  if (granuleSearchParams.granuleId) {
+    dynamoSearchType = 'query';
+    extraSearchParams = {
+      KeyConditionExpression: 'granuleId = :granuleId',
+      ExpressionAttributeValues: {
+        ':granuleId': granuleSearchParams.granuleId,
+      },
+    };
+  } else if (granuleSearchParams.collectionId) {
+    dynamoSearchType = 'query';
+    extraSearchParams = {
+      IndexName: 'collectionId-granuleId-index',
+      KeyConditionExpression: 'collectionId = :collectionId',
+      ExpressionAttributeValues: {
+        ':collectionId': granuleSearchParams.collectionId,
```

Comment:
I opted to support `collectionId` (concatenation of collection name/version) as the supported filter value since that is how the value shows up in Dynamo, so they can copy it straight from Dynamo records into the payload for this migration